plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts

```diff
@@ -123,24 +121,30 @@ export class DefaultCatalogCollatorFactory implements DocumentCollatorFactory {
 
       for (const entity of entities) {
         yield {
-          title: entity.metadata.title ?? entity.metadata.name,
+          ...this.entityTransformer(entity),
```

Comment:
Title and text are required fields. How would we prevent people from not passing them?
Are we relying on the type errors only? Perhaps they could be optional and we could define a default title and text if not passed.

Reply:
I guess we can rely on type errors, because otherwise we will need to duplicate all type checks on execution as with typescript you can do whatever you want, like
```ts
class A {}
const a = new A();
const b = a as unknown as string; // TS thinks b is string
```